Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -561,18 +568,26 @@ TextureCacheBase::TCacheEntry* TextureCacheBase::Load(const u32 stage)
 
   // TODO: This doesn't hash GB tiles for preloaded RGBA8 textures (instead, it's hashing more data
   // from the low tmem bank than it should)
-  base_hash = GetHash64(src_data, texture_size, g_ActiveConfig.iSafeTextureCache_ColorSamples);
+  bool done_hash = false;
   u32 palette_size = 0;
   if (isPaletteTexture)
-  {
     palette_size = TexDecoder_GetPaletteSize(texformat);
-    full_hash = base_hash ^ GetHash64(&texMem[tlutaddr], palette_size,
-                                      g_ActiveConfig.iSafeTextureCache_ColorSamples);
-  }
-  else
-  {
-    full_hash = base_hash;
-  }
+  auto DoHash = [&]() {
```

Comment:
Do you think this functions fits to the scope of the entry itself? It already knows if it is locked, so it should be able to rehash as well.

Reply:
There's probably a better/cleaner way of doing this. Basically, I wanted to only have to calculate the hash for the incoming texture if an entry with the matching address requires a rehash. This avoids the GPU thread from reading the guest memory if it is locked too, which would otherwise trigger a fault/flush.